### ISSUE TITLE
Add support for :string in FFI::Struct layout

### ISSUE THREAD

**Issue by tmwangi** [ffi/ffi#455]
At the moment, one cannot have a variable of the type :string in an FFI::Struct layout. Just wondering if there are plans to support this in the future.

**Comment by tduehr** [comment]
You're expecting `:string` to be a pointer, correct?

**Comment by tmwangi** [comment]
Yes. So every time I have had to use in the struct I have had to call ::FFI::MemoryPointer.from_string(s). And this allocates memory and stuff. Just wondering if there is a way to pass s as a :string directly without having to allocate memory. (Use the char array in ruby string object or something). But maybe I am missing something.
